### PR TITLE
HTCONDOR-2672 64 bit time_t

### PR DESCRIPTION
**Pull request by GregThain**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_utils/cron_job_params.cpp [hunk 1]
```diff
@@ -232,7 +232,7 @@ CronJobParams::InitPeriod( const std::string &param_period )
 		return false;
 	} else {
 		char	modifier = 'S';
-		int		num = sscanf( param_period.c_str(), "%d%c",
+		int		num = sscanf( param_period.c_str(), "%ld%c",
```
Comment (JaimeFrey) [inline src/condor_utils/cron_job_params.cpp:235]: time_t is a long long on Windows. I think you need to store the value in a local long long variable, then assign into m_period after the sscanf().

File: src/condor_daemon_core.V6/condor_timer_manager.h [hunk 2]
```diff
@@ -68,17 +66,16 @@ typedef void    (Service::*Releasecpp)(void*);
 #define TimerReleasecpp Releasecpp
 
 // This value, passed for "when", will cause the timer to never expire
-const unsigned	TIMER_NEVER = 0xffffffff;
-
-const time_t TIME_T_NEVER	= 0x7fffffff;
+const time_t TIMER_NEVER	 = -1;
+const time_t TIME_T_NEVER	= -1;
```
Comment (JaimeFrey) [inline src/condor_daemon_core.V6/condor_timer_manager.h:70]:
```suggestion
const time_t TIMER_NEVER	 = std::numeric_limits<time_t>::max();
const time_t TIME_T_NEVER	= std::numeric_limits<time_t>::max();
```
These need to be positive values.

**Review by JaimeFrey — CHANGES_REQUESTED** [review]
TimerManager::ResetTimer() isn't prepared for either `when` or `period` to be TIMER_NEVER that's not the old INT_MAX-ish value.

**Comment by GregThain** [comment]
@JaimeFrey changes look good to me, and a new coverity run shows no new problems.

**Review by JaimeFrey — APPROVED** [review]
LGTM

**Pull request merged**